thirdeye/thirdeye-hadoop/src/main/java/com/linkedin/thirdeye/hadoop/push/DeleteOverlappingSegmentsInPinot.java

```diff
@@ -0,0 +1,129 @@
+package com.linkedin.thirdeye.hadoop.push;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.TreeSet;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.helix.AccessOption;
+import org.apache.helix.BaseDataAccessor;
+import org.apache.helix.HelixDataAccessor;
+import org.apache.helix.PropertyKey;
+import org.apache.helix.PropertyKey.Builder;
+import org.apache.helix.ZNRecord;
+import org.apache.helix.manager.zk.ZKHelixDataAccessor;
+import org.apache.helix.manager.zk.ZNRecordSerializer;
+import org.apache.helix.manager.zk.ZkBaseDataAccessor;
+import org.apache.helix.manager.zk.ZkClient;
+import org.apache.helix.model.ExternalView;
+import org.apache.helix.model.IdealState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DeleteOverlappingSegmentsInPinot {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DeleteOverlappingSegmentsInPinot.class);
+
+  public static void main(String[] args) throws Exception {
+    String zkUrl = args[0];
+    String zkCluster = args[1];
+    String tableName = args[2];
+    deleteOverlappingSegments(zkUrl, zkCluster, tableName);
+  }
+
+  private static IdealState computeNewIdealStateAfterDeletingOverlappingSegments(HelixDataAccessor helixDataAccessor, PropertyKey idealStatesKey) {
+    IdealState is = helixDataAccessor.getProperty(idealStatesKey);
+    // compute existing DAILY segments
+    Set<String> daysWithDailySegments = new HashSet<>();
+    for (String segmentName : is.getPartitionSet()) {
+      LOG.info("Segment Name : {}", segmentName);
+      if (segmentName.indexOf("DAILY") > -1) {
+        String[] splits = segmentName.split("_");
+        String endDay = splits[splits.length - 2].substring(0, "yyyy-mm-dd".length());
```

Comment:
Since same strings literal is used at more than one places, we should extract as static final and reuse.